### ISSUE TITLE
Provide simple get()

### ISSUE THREAD

**Issue by dsvoronin** [Jearil/SimpleNoSQL#8]
In addition to retrieve() method, it will be great if i can handle all by myself. Simple synchronous call. Useful when i want to dispatch db calls through my own executor for example.

**Comment by appmagnetics** [comment]
Hi, you are asking "who will ever want to shoot his healthy feet and block main thread to retrieve data"? Well, I want to load list of accounts on app start. I want to avoid blocked reading (although it would be convenient to me just have a syncron load() call), so I'm trying to start retrieval normally - async way - and to syncronize on it later in start up process. But it does not work so far:

I'm using CountDownLatch() as in your examples and count it down in the RetrievalCallback. 
At the same time Main Thread waits until same latch. But the app goes into deadlock like this. I assume that RetreivalCallback code is executed in the Main Thread, which is blocked, that's why I never see callback executed. 

Could you advice some other way of syncronizing?

**Comment by Jearil** [comment]
I'm not really sure what you're asking here. Most apps will want to send a request to retrieve data and just use a callback to populate the data when it's retrieved, probably updating UI as appropriate. You won't want to block the UI thread as that will give you ANR errors. It's often best just to provide a loading indicator such as a progress bar if you're loading a lot of data that takes a noticeable amount of time.

**Comment by appmagnetics** [comment]
We have a small chunk of data - like 10 records, it does not deserve loading UI and it must be loaded on app start. We have used Sugar component before and when you issue SQL request there you get the data synchronously. So I see now in your component the concept is the opposite - default handling is async. So far so good, I just still could not get why Retrieval execution gets blocked if I wait in UI thread - I expect Retrieval callback to be executed in separate thread always. For now we have found a workaround and syncronize start process differently, so no further help required on this topic, thank you!
